staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1005,3 +1005,88 @@ const (
+	// may reject unrecognized values.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,1,opt,name=apiVersion"`
+	// Fields TODO
+	Fields FieldSet `json:"fields,omitempty" protobuf:"bytes,2,opt,name=fields,casttype=FieldSet"`
+}
+
+// FieldSet identifies a set of fields.
+type FieldSet struct {
+	// Members lists fields that are part of the set.
+	Members []FieldPathElement `json:"members" protobuf:"bytes,1,rep,name=members"`
+	// Children lists child fields which themselves have children that are
+	// members of the set. Appearance in this list does not imply membership.
+	// Note: this is a tree, not an arbitrary graph.
+	Children []FieldSetNode `json:"children" protobuf:"bytes,2,rep,name=children"`
+}
+
+// FieldPathElement describes how to select a child field given a containing object.
+type FieldPathElement struct {
+	// Exactly one of the following fields should be non-nil.
+	// FieldName selects a single field from a map (reminder: this is also
+	// how structs are represented). The containing object must be a map.
+	// +optional
+	FieldName *string `json:"fieldName,omitempty" protobuf:"bytes,1,opt,name=fieldName"`
+	// Key selects the list element which has fields matching those given.
+	// The containing object must be an associative list with map typed
+	// elements.
+	// +optional
+	Key []FieldValuePair `json:"key,omitempty" protobuf:"bytes,2,rep,name=key"`
+	// Value selects the list element with the given value. The containing
+	// object must be an associative list with a primitive typed element
+	// (i.e., a set).
+	// +optional
+	Value *Value `json:"value,omitempty" protobuf:"bytes,3,opt,name=value"`
+	// Index selects a list element by its index number. The containing
+	// object must be an atomic list.
+	// +optional
+	Index *int32 `json:"index,omitempty" protobuf:"varint,4,opt,name=index"`
+}
+
+// FieldSetNode is a pair of FieldPathElement / FieldSet, for the purpose of expressing
+// nested set membership.
+type FieldSetNode struct {
+	// PathElement TODO
+	PathElement FieldPathElement `json:"pathElement" protobuf:"bytes,1,opt,name=pathElement"`
+	// Set TODO
+	Set *FieldSet `json:"set" protobuf:"bytes,2,opt,name=set"`
+}
+
+// FieldValuePair is an individual key-value pair.
+type FieldValuePair struct {
+	// Name TODO
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// Value TODO
+	Value Value `json:"value" protobuf:"bytes,2,opt,name=value"`
+}
+
+// A Value is an object; it corresponds to an 'atom' in the schema.
+type Value struct {
+	// Exactly one of the following fields should be set.
+	// Float TODO
+	// +optional
+	Float *float64 `json:"float,omitempty" protobuf:"bytes,1,opt,name=float"`
+	// Int TODO
+	// +optional
+	Int *int32 `json:"int,omitempty" protobuf:"varint,2,opt,name=int"`
+	// String TODO
+	// +optional
+	Str *string `json:"str,omitempty" protobuf:"bytes,3,opt,name=str"`
```

Comment:
String

Reply:
`String` caused problems with protobuf

Reply:
Really?

ಠ_ಠ

Reply:
I think it needs to reserve the .String() function to satisfy some interface

Reply:
That's super unfortunate. I'll have to think about that. Abbreviations are disallowed in our API with very few exceptions. :/

Reply:
can make the json name "string", though I wonder if that's going to trigger keyword errors in some of the openapi generated clients

Reply:
intValue, floatValue, stringValue are stuttery, but non-keyword-infringing